core/src/main/java/com/graphhopper/storage/CHGraph.java

```diff
@@ -82,4 +83,8 @@
      * @return the number of original edges in this graph (without shortcuts)
      */
     int getOriginalEdges();
+
+    NodeOrderingProvider getNodeOrderingProvider();
+
+    boolean isFrozen();
```

Comment:
I just had to put it out of the constructor because in many cases PCH is created before the graph is 'frozen'. So I moved the check to `doSpecificWork()`. Actually `isFrozen()` is probably not a good name either, because the graph is not 'frozen', but rather nodesCH is initialized so we can start adding shortcuts and set CH levels. So I renamed it to 'isReadyForContraction' or what do you think ?